### ISSUE TITLE
False Positives in 2.6.0

### ISSUE THREAD

**Issue by andig** [golangci/golangci-lint#6159]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Official GitHub Action

### Description of the problem

Since 2.6.0 I'm seing various false positives: https://github.com/evcc-io/evcc/actions. I suspect golangci-lint since the positives include `vet` and `gci`, independent of any code changes. Latest example is https://github.com/evcc-io/evcc/actions/runs/18933135220/job/54053788167 where generated files are linted.

I did not try linters standalone since multiple affected and no code changes.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by andig** [comment]
Local run:

```
golangci-lint cache clean
golangci-lint run -v
INFO golangci-lint has version 2.6.0 built with go1.25.3 from fb09c37 on 2025-10-29T19:28:07Z 
INFO [config_reader] Config search paths: [./ /Users/a25058/htdocs/evcc /Users/a25058/htdocs /Users/a25058 /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO maxprocs: Leaving GOMAXPROCS=12: CPU quota undefined 
INFO [goenv] Read go env for 5.405834ms: map[string]string{"GOCACHE":"/Users/a25058/Library/Caches/go-build", "GOROOT":"/opt/homebrew/Cellar/go/1.25.3/libexec"} 
INFO [lintersdb] Active 19 linters: [durationcheck gci gofmt goimports goprintffuncname govet importas ineffassign makezero misspell nolintlint rowserrcheck sqlclosecheck staticcheck tparallel unconvert unused wastedassign whitespace] 
INFO [loader] Go packages loading at mode 8767 (deps|exports_file|files|compiled_files|imports|name|types_sizes) took 1.738201125s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 55.024667ms 
INFO [linters_context] importas settings found, but no aliases listed. List aliases under alias: key. 
INFO [linters_context/goanalysis] analyzers took 2m49.447069667s with top 10 stages: goimports: 1m6.930374583s, wastedassign: 38.687091667s, unused: 10.831316127s, gci: 8.086298327s, buildssa: 5.540349925s, gofmt: 4.878241127s, misspell: 4.24540513s, unconvert: 4.23386287s, printf: 3.703181733s, ctrlflow: 3.400680012s 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "third_party$" 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "builtin$" 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "examples$" 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "third_party$", Linters: "gci, gofmt, goimports"] 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "builtin$", Linters: "gci, gofmt, goimports"] 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "examples$", Linters: "gci, gofmt, goimports"] 
INFO [runner] Issues before processing: 70, after processing: 0 
INFO [runner] Processors filtering stat (in/out): path_absoluter: 70/70, exclusion_paths: 70/70, invalid_issue: 70/70, path_relativity: 70/70, generated_file_filter: 70/0, cgo: 70/70, filename_unadjuster: 70/70 
INFO [runner] processing took 3.432961ms with stages: generated_file_filter: 3.328875ms, exclusion_paths: 57.834µs, path_relativity: 22.292µs, exclusion_rules: 11.334µs, invalid_issue: 3.208µs, sort_results: 2.708µs, cgo: 1.667µs, path_absoluter: 1.041µs, max_same_issues: 750ns, nolint_filter: 750ns, fixer: 501ns, filename_unadjuster: 417ns, source_code: 375ns, path_prettifier: 250ns, uniq_by_line: 209ns, diff: 207ns, path_shortener: 167ns, max_from_linter: 167ns, max_per_file_from_linter: 125ns, severity-rules: 84ns 
INFO [runner] linters took 17.197287166s with stages: goanalysis_metalinter: 17.193784125s 
0 issues.
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 191 samples, avg is 1117.5MB, max is 1542.1MB 
INFO Execution took 19.004169167s                 
```

**Comment by andig** [comment]
However, I'm still consuming >8G caches and build is still using cache:

```
Run actions/cache@v4
Cache hit for: Linux-go-lint-0ab5c32504fcb99ae580b215740023f6228a9ffac06c86f63eb7666e69ca8e0c
Cache Size: ~843 MB (883704271 B)
/usr/bin/tar -xf /home/runner/work/_temp/97cff2b0-6260-4429-929b-0cd94afbd980/cache.tzst -P -C /home/runner/work/evcc/evcc --use-compress-program unzstd
Cache restored successfully
Cache restored from key: Linux-go-lint-0ab5c32504fcb99ae580b215740023f6228a9ffac06c86f63eb7666e69ca8e0c
```

This looks like a breakage on GH?
